src/ota.rs

```diff
@@ -220,16 +220,43 @@ impl EspOta {
     pub fn initiate_update(&mut self) -> Result<&mut EspOtaUpdate, EspError> {
         self.check_read()?;
 
-        let partition = unsafe { esp_ota_get_next_update_partition(ptr::null()) };
+        // TODO: The documentation of `esp_ota_get_next_update_partition` states, that the returned
+        // pointer may be a null pointer, which indicates that:
+        // "indicates invalid OTA data partition, or that no eligible OTA app slot partition was found."
+        //
+        // This should probably be handled in here and throw a proper EspError.
+        let update_partition = unsafe { esp_ota_get_next_update_partition(ptr::null()) };
 
-        let mut handle: esp_ota_handle_t = Default::default();
+        let mut update_handle: esp_ota_handle_t = Default::default();
 
-        esp!(unsafe { esp_ota_begin(partition, OTA_SIZE_UNKNOWN as usize, &mut handle) })?;
+        esp!(unsafe {
+            esp_ota_begin(
+                update_partition,
+                OTA_SIZE_UNKNOWN as usize,
+                &mut update_handle,
+            )
+        })?;
+
+        self.0 = Some(EspOtaUpdate {
+            update_partition,
+            update_handle,
+            discarded: false,
+        });
 
-        self.0.update_partition = partition;
-        self.0.update_handle = handle;
+        Ok(self.0.as_mut().unwrap())
+    }
+
+    /// Get a handle to the [EspOtaUpdate].
+    /// This is `None`, if [Self.initiate_update] hasn't been called yet.
+    pub fn get_update(&mut self) -> Option<&mut EspOtaUpdate> {
```

Comment:
Check if `update_partition` is not null and then return `Some(...)` else `None`.